File: Backtracking/javascript/1182.js
const input = require("fs").readFileSync("/dev/stdin").toString().split("\n");
let result = 0;
const target = Number(input[0].split(" ")[1]);
const arr = input[1].split(" ").map(Number);

const search = (num, idx) => {
  if (num === target) {
    result++;
  }
  for (let i = idx; i < arr.length; i++) {
    search(num + arr[i], i + 1);
  }
};
search(0 || null, 0);

console.log(result);

File: Backtracking/python/1182.py
n,s = map(int,input().split())
arr = list(map(int,input().split()))
start = 0
result = 0

def search(num,idx,target):
  global result
  if num == target:
    result += 1
  for i in range(idx,len(arr)):
    search(num+arr[i],i+1,target)
if s==0:
  result = -1
search(0,0,s)
print(result)